[do not merge] update samples to take Microsoft Identity Web as a nuget package

jennyf19 opened the pull request:
- I have tested all the samples, but would appreciate another set or two of eyes. Not sure if I've caught all the ReadMe updates. Appreciate any testing you can do as well. thanks.

jmprieur commented on 5-WebApp-AuthZ/5-2-Groups/Startup.cs:
```diff
@@ -36,16 +36,13 @@ public void ConfigureServices(IServiceCollection services)
                     //    // Use the groups claim for populating roles
                     //    options.TokenValidationParameters.RoleClaimType = "groups";
-                });
+                }, options => { });
```
the second options should probably do the binding as well? `Configuration.Bind("AzureAD", options);` ?

jmprieur commented on 2-WebApp-graph-user/2-3-Multi-Tenant/Startup.cs:
```diff
@@ -106,7 +85,7 @@ public void ConfigureServices(IServiceCollection services)
                         return Task.FromResult(0);
                     };
-                });
+                }, option => { });
```
the second options callback should probably do the binding as well? `Configuration.Bind("AzureAD", options);` ?

jmprieur commented on 2-WebApp-graph-user/2-3-Multi-Tenant/Startup.cs:
```diff
@@ -78,7 +57,7 @@ public void ConfigureServices(IServiceCollection services)
+                .AddSignIn(options =>
                 {
                     Configuration.Bind("AzureAD", options);
 
```
We also need to check what happen here. I'm not sure how the overrides of the events will be handled

jennyf19 replied:
we can sync on this

---
In reply to: [406347523](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/pull/331#discussion_r406347523) [](ancestors = 406347523)

jmprieur commented on 5-WebApp-AuthZ/5-2-Groups/WebApp-OpenIDConnect-DotNet.csproj:
```diff
@@ -19,11 +19,7 @@
-  <ItemGroup>
-    <ProjectReference Include="..\..\Microsoft.Identity.Web.UI\Microsoft.Identity.Web.UI.csproj" />
-    <ProjectReference Include="..\..\Microsoft.Identity.Web\Microsoft.Identity.Web.csproj" />
+    <PackageReference Include="Microsoft.Identity.Web.UI" Version="0.1.0-localbuild" />
```
Oh, I understand Microsoft.Identityu.Web.UI also references Microsoft.Identity.Web ...

jennyf19 replied:
fixing these. thanks.

---
In reply to: [406348152](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/pull/331#discussion_r406348152) [](ancestors = 406348152)

jennyf19 commented on 1-WebApp-OIDC/1-1-MyOrg/Startup.cs:
```diff
@@ -35,8 +35,8 @@ public void ConfigureServices(IServiceCollection services)
             //services.AddSignIn(Configuration);
-            services.AddAuthentication(OpenIdConnectDefaults.AuthenticationScheme)
-                .AddSignIn("AzureAD", Configuration, options => Configuration.Bind("AzureAD", options));
+            services.AddAuthentication(OpenIdConnectDefaults.AuthenticationScheme);
```
thanks. had fixed this but the changes didn't get pushed up last night. will do. :)

jmprieur commented on 2-WebApp-graph-user/2-3-Multi-Tenant/Startup.cs:
```diff
@@ -56,11 +56,8 @@ public void ConfigureServices(IServiceCollection services)
-                .AddSignIn(options =>
+            services.AddSignIn(options =>
                 {
-                    Configuration.Bind("AzureAD", options);
```
I think we still need `Configuration.Bind("AzureAD", options);` both here and in line 88.

jennyf19 replied:
I've fixed it. thx

---
In reply to: [406889283](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/pull/331#discussion_r406889283) [](ancestors = 406889283)

jennyf19 commented:
i think addressed all the comments. thank you.

---
In reply to: [390969734](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/pull/331#issuecomment-390969734) [](ancestors = 390969734)

pmaytak commented on 1-WebApp-OIDC/1-1-MyOrg/WebApp-OpenIDConnect-DotNet.csproj:
```diff
@@ -25,12 +25,9 @@
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
     <PackageReference Include="Microsoft.Extensions.Logging.Debug" Version="3.0.0" />
+    <PackageReference Include="Microsoft.Identity.Web" Version="0.1.0-localbuild" />
```
Hmm, should version be `0.1.0-preview` or something?

jennyf19 replied:
I know. I wanted to test everything w/the nuget package w/the current release build. we will have to run a new package w/the correct name.

pmaytak commented on 5-WebApp-AuthZ/5-2-Groups/Startup.cs:
```diff
@@ -36,15 +36,15 @@ public void ConfigureServices(IServiceCollection services)
             // Sign-in users with the Microsoft identity platform
-            services.AddAuthentication(OpenIdConnectDefaults.AuthenticationScheme)
-                .AddSignIn("AzureAD", Configuration, options =>
+            services.AddSignIn(options =>
```
Missing `Configuration.Bind("AzureAD", options);` in the first options action?

jennyf19 commented:
@TiagoBrenck I ran the [build again](https://identitydivision.visualstudio.com/IDDP/_build/results?buildId=508771&view=logs&j=275f1d19-1bd8-5591-b06b-07d489ea915a&t=368b987b-4e63-5984-5006-1d171be4473f) and things are passing. did you address all the comments above in your earlier PR? thx.

jennyf19 commented on 1-WebApp-OIDC/1-1-MyOrg/README.md:
````diff
@@ -164,8 +164,7 @@ cd "1-WebApp-OIDC\1-1-MyOrg"
      ```CSharp
-      services.AddAuthentication(OpenIdConnectDefaults.AuthenticationScheme)
-              .AddSignIn("AzureAd", Configuration, options => Configuration.Bind("AzureAd", options));
+      services.AddSignIn(Configuration);
````
yes, we probably should. can you do that?

---
In reply to: [406857441](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/pull/331#discussion_r406857441) [](ancestors = 406857441)

jennyf19 commented:
this was addressed right?

---
In reply to: [391557054](https://github.com/Azure-Samples/active-directory-aspnetcore-webapp-openidconnect-v2/pull/331#issuecomment-391557054) [](ancestors = 391557054)

jennyf19 commented:
@jmprieur this sample is ready to merge. we will let you do a final pass and merge when ready. @TiagoBrenck  has a few points to discuss and will do so via email. :)
